x-pack/plugins/reporting/server/lib/screenshots/wait_for_visualizations.ts

```diff
@@ -11,10 +11,23 @@ import { HeadlessChromiumDriver } from '../../browsers';
 import { LayoutInstance } from '../layouts';
 import { CONTEXT_WAITFORELEMENTSTOBEINDOM } from './constants';
 
-type SelectorArgs = Record<string, string>;
+interface CompletedItemsCountParameters {
+  context: string;
+  count: number;
+  renderCompleteSelector: string;
+}
 
-const getCompletedItemsCount = ({ renderCompleteSelector }: SelectorArgs) => {
-  return document.querySelectorAll(renderCompleteSelector).length;
+const getCompletedItemsCount = ({
+  context,
+  count,
+  renderCompleteSelector,
+}: CompletedItemsCountParameters) => {
+  const { length } = document.querySelectorAll(renderCompleteSelector);
+
+  // eslint-disable-next-line no-console
+  console.debug(`evaluate ${context}: waitng for ${count} elements, got ${length}.`);
```

Comment:
Note: this function is evaluated in the headless browser, so the `console.debug` is perfectly fine:
![image](https://user-images.githubusercontent.com/908371/140996162-be4e7f4e-231b-4006-a004-93fbd03d3c20.png)